### PR TITLE
Rheology module for viscosity multiplication

### PR DESCRIPTION
**Pull request by naliboff**
This PR supersedes #3184 by adding a new rheology module that multiplies the viscosity of each composition by a user-specified value.

* [X] I have followed the [instructions for indenting my code](../blob/master/CONTRIBUTING.md#making-aspect-better).
* [X] I have tested my new feature locally to ensure it is correct.
* [X] I have [created a testcase](http://www.math.clemson.edu/~heister/manual.pdf#sec%3Awriting_tests) for the new feature/benchmark in the [tests/](../blob/master/tests/) directory.
* [X] I have added a changelog entry in the [doc/modules/changes](../blob/master/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

File: include/aspect/material_model/rheology/constant_viscosity_prefactors.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+        private:
+          /**
+           * The constant viscosity prefactors, which are read in
+          * from the input file by the parse_parameters() function.
+          * The total number of prefactors will be equal to one
+          * plus the number of compositional fields. The prefactor
+          * for a given compositional field is multiplied with a
+          * viscosity value provided by the material model, which
+          * is then returned to the material model.
```
Comment (gassmoeller) [inline include/aspect/material_model/rheology/constant_viscosity_prefactors.h:73]: These lines are missing a leading space.

File: source/material_model/rheology/constant_viscosity_prefactors.cc [hunk 2]
```diff
@@ -0,0 +1,93 @@
+/*
+  Copyright (C) 2019 by the authors of the ASPECT code.
```
Comment (gassmoeller) [inline source/material_model/rheology/constant_viscosity_prefactors.cc:2]: 2020

File: source/material_model/rheology/constant_viscosity_prefactors.cc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace Rheology
+    {
+      template <int dim>
+      ConstantViscosityPrefactors<dim>::ConstantViscosityPrefactors ()
+      {}
```
Comment (gassmoeller) [inline source/material_model/rheology/constant_viscosity_prefactors.cc:37]: missing initialization to NaN as described in the header

Comment (naliboff) [reply source/material_model/rheology/constant_viscosity_prefactors.cc:37]: Ahh, I meant to remove that documentation from the header file. I'm actually not sure what variable is being initialized as NaN in `constant_visosity`? Is this the private member variable `viscosity`?

Comment (naliboff) [reply source/material_model/rheology/constant_viscosity_prefactors.cc:37]: @gassmoeller - Thanks for the review! I addressed the two other comments and this one by removing the reference to the `NaN initialization`, but let me know if that  initialization should be included.

File: include/aspect/material_model/rheology/constant_viscosity_prefactors.h [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    namespace Rheology
+    {
+      template <int dim>
```
Comment (gassmoeller) [inline include/aspect/material_model/rheology/constant_viscosity_prefactors.h:36]: Forgot to mention this in the first round of reviews: Please add documentation for the class here.

File: source/material_model/rheology/constant_viscosity_prefactors.cc (hunk 3 above)
Comment (gassmoeller) [reply source/material_model/rheology/constant_viscosity_prefactors.cc:37]: This this fine. We do not know the length of the vector anyway.

File: include/aspect/material_model/rheology/constant_viscosity_prefactors.h (hunk 4 above)
Comment (naliboff) [reply include/aspect/material_model/rheology/constant_viscosity_prefactors.h:41]: Done and I squashed this into the most recent commit.

**Comment by naliboff** [comment]
@gassmoeller - I think this is good to go unless you would like further edits to the class documentation. Thanks again for the review!

**Pull request merged**